### ISSUE TITLE
Error on product detail page with excluding tax

### ISSUE THREAD

**Issue by drahreg** [firegento/firegento-magesetup#137]
## Description

If you set set

* System -> Configruation -> Sales -> Tax -> Calculation Settings -> Catalog Prices
* System -> Configruation -> Sales -> Tax -> Price Display Settings -> Display Product Prices In Catalog

both to *Excluding Tax* an error occurs on product detail page

## Error

Fatal error: Call to a member function getId() on a non-object in path/to/magento/app/code/core/Mage/Tax/Model/Calculation.php on line 303

## Used Versions

Magento: 1.9.1.0
FireGento_MageSetup: 2.2.1

## Function in which the error occurs
```php
/**
 * Get cache key value for specific tax rate request
 *
 * @param   $request
 * @return  string
 */
protected function _getRequestCacheKey($request)
{
    $key = $request->getStore() ? $request->getStore()->getId() . '|' : '';
    $key .= $request->getProductClassId() . '|' . $request->getCustomerClassId() . '|'
        . $request->getCountryId() . '|' . $request->getRegionId() . '|' . $request->getPostcode();
    return $key;
}
```

## The $request object

I put a `var_dump($request->debug());` to the top of the function and the result is

```php
array (size=6)
  'country_id' => string 'US' (length=2)
  'region_id' => string '0' (length=1)
  'postcode' => string '*' (length=1)
  'store' => string '3' (length=1)
  'customer_class_id' => string '10' (length=2)
  'product_class_id' => string '2' (length=1)
```

If i change

*System -> Configruation -> Sales -> Tax -> Calculation Settings -> Catalog Prices* from *Excluding Tax* to *Including Tax* 

or

*System -> Configruation -> Sales -> Tax -> Price Display Settings -> Display Product Prices In Catalog* from *Excluding Tax* to *other value*

it works and the result is

```php
array (size=6)
  'country_id' => string 'US' (length=2)
  'region_id' => string '0' (length=1)
  'postcode' => string '*' (length=1)
  'store (Mage_Core_Model_Store)' => 
    array (size=12)
      'store_id' => string '3' (length=1)
      'code' => string 'german' (length=6)
      'website_id' => string '1' (length=1)
      'group_id' => string '1' (length=1)
      'name' => string 'German' (length=6)
      'sort_order' => string '0' (length=1)
      'is_active' => string '1' (length=1)
      'locale_code' => string 'de_DE' (length=5)
      'home_url' => string 'http://magento1910.dev/?___store=german' (length=39)
      'available_currency_codes' => 
        array (size=1)
          0 => string 'EUR' (length=3)
      'base_currency (Mage_Directory_Model_Currency)' => 
        array (size=1)
          'currency_code' => string 'EUR' (length=3)
      'current_currency (Mage_Directory_Model_Currency)' => 
        array (size=1)
          'currency_code' => string 'EUR' (length=3)
  'customer_class_id' => string '10' (length=2)
  'product_class_id' => string '2' (length=1)
```

## How to reproduce?

* Module Firegento_Magesetup is installed
* Change value of *System -> Configruation -> Sales -> Tax -> Calculation Settings -> Catalog Prices* to *Excluding Tax*
* Change value of *System -> Configruation -> Sales -> Tax -> Price Display Settings -> Display Product Prices In Catalog* to *Excluding Tax*
* Disable Cache
* Open product detail site of an taxable product

or 

* Install magento in version 1.9.1.0 with sample data
* Disable cache
* Open Browser and go to http://[vhost]/accessories/eyewear/aviator-sunglasses.html (It is an simple taxable product)
* install module FireGento_MageSetup
* Refresh page and the error occurs

## Notes

* If i uninstall the firegento_magesetup module the error is gone
* The function `_getRequestCacheKey()` seems not to be called in clean magento installation

**Comment by torstenkerneck** [comment]
I got the same problem. Quickfix to handle this issue was to change the function at Calculation.php. It's not the clean way, but helps in the meantime. I hope the issue get solved quickly.
```javascript
    protected function _getRequestCacheKey($request)
    {
        $store = $request->getStore(); // temporary buffering
        $key  = ( $store ? ( is_object( $store ) ? $store->getId() : $store ).'|' : '' ); // solve issue
        $key .= $request->getProductClassId() . '|' . $request->getCustomerClassId() . '|'
            . $request->getCountryId() . '|' . $request->getRegionId() . '|' . $request->getPostcode();
        return $key;
   }
```

**Comment by sprankhub** [comment]
This should be fixed in version 2.2.2 and can therefore be closed, see https://github.com/firegento/firegento-magesetup/commit/2fda3c6e197b2e751ffda2fa1954357a60798f6b
